### ISSUE TITLE
geckodriver executable not found if it is in a folder with other executables

### ISSUE THREAD

**Issue by MarioStoilov** [SeleniumHQ/selenium#2271]
## Meta -
OS:  windows 10
Selenium Version:  2.53.1
Browser:  Firefox (version irrelevant)

## Steps to reproduce -
put the geckodriver executable in a folder with other exes and try to start the webdriver with marionette

**Comment by MarioStoilov** [comment]
Actually, it doesn't find the executable at all even if it is in the folder alone. Debugged the issue and found the problem.

in platform.rb
```
def find_binary(*binary_names)
        paths = ENV['PATH'].split(File::PATH_SEPARATOR)
        binary_names.map! { |n| "#{n}.exe" } if windows?

        binary_names.each do |binary_name|
          paths.each do |path|
            exe = Dir.glob(fFile.join(path, binary_name)).first 
            next unless exe
            return exe if File.executable?(exe)
          end
        end
```

According to Dir.glob documentation:

>     # <code> \\ </code>::
>     #   Escapes the next metacharacter.
>     # 
>     #   Note that this means you cannot use backslash on windows as part of a
>     #   glob, i.e.  <code>Dir["c:\\foo*"]</code> will not work, use
>     #   <code>Dir["c:/foo*"]</code> instead.

changing the method to:
```
      def find_binary(*binary_names)
        paths = ENV['PATH'].split(File::PATH_SEPARATOR)
        binary_names.map! { |n| "#{n}.exe" } if windows?

        binary_names.each do |binary_name|
          paths.each do |path|
            full_path = File.join(path, binary_name).gsub! '\\', '/'
            exe = Dir.glob(full_path).first
            next unless exe
            return exe if File.executable?(exe)
          end
        end
```

fixes the problem

**Comment by titusfortner** [comment]
Fixed & released in selenium-webdriver-2.53.2.gem
Thank you.
